src/ucp/tag/eager_snd.c

```diff
@@ -32,8 +32,6 @@ ucp_tag_pack_eager_common(ucp_request_t *req, void *dest,
     packed_length = ucp_dt_pack(req->send.ep->worker, req->send.datatype,
                                 req->send.mem_type, dest, req->send.buffer,
                                 &req->send.state.dt, length);
-    ucs_assert((single && (packed_length == req->send.length)) ||
```

Comment:
yes, single can be removed. I've removed the assert because a protocol can start as multi (and pass single=0) but eventually send only one packet, if max_bcopy is larger than UCP_MIN_BCOPY